### ISSUE TITLE
NullPointerException when using TAB key in CheckComboBox

### ISSUE THREAD

**Issue by patrick0210** [controlsfx/controlsfx#1297]
When checking out the 9.0.0 branch and running the sample application, a NullPointerException is thrown in the CheckComboBox example. It can be reproduced by clicking on the "CheckComboBox" or "CheckCombobox with data objects", then selecting an arbitrary element and finally hitting the TAB key. The correct behavior according to the source code would be a jump to the next control. The actual behavior is only the output of an exception.

The logged exception:

```
Exception in thread "JavaFX Application Thread" java.lang.NullPointerException
        at javafx.controls/javafx.scene.control.skin.VirtualFlow$3.findOwnerCell(VirtualFlow.java:645)
        at javafx.controls/javafx.scene.control.skin.VirtualFlow$3.select(VirtualFlow.java:619)
        at javafx.graphics/com.sun.javafx.scene.traversal.TraversalEngine.select(TraversalEngine.java:104)
        at javafx.graphics/com.sun.javafx.scene.traversal.TopMostTraversalEngine.trav(TopMostTraversalEngine.java:77)
        at javafx.graphics/javafx.scene.Scene.traverse(Scene.java:2092)
        at javafx.graphics/javafx.scene.Node.traverse(Node.java:8323)
        at javafx.graphics/javafx.scene.Node$1.traverse(Node.java:516)
        at javafx.graphics/com.sun.javafx.scene.NodeHelper.traverse(NodeHelper.java:236)
        at javafx.controls/com.sun.javafx.scene.control.behavior.FocusTraversalInputMap.traverse(FocusTraversalInputMap.java:98)
        at javafx.controls/com.sun.javafx.scene.control.behavior.FocusTraversalInputMap.traverseNext(FocusTraversalInputMap.java:138)
        at javafx.controls/com.sun.javafx.scene.control.behavior.FocusTraversalInputMap.lambda$static$4(FocusTraversalInputMap.java:53)
        at javafx.controls/com.sun.javafx.scene.control.inputmap.InputMap.handle(InputMap.java:274)
        at javafx.base/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
        at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
```

**Comment by abhinayagarwal** [comment]
I cannot reproduce this. What JavaFX version are you running against?

**Comment by patrick0210** [comment]
Interesting. I tested it with JavaFX 11.0.2 and 14.0.2 and I use openJDK 11. A NullPointerException gets logged only if an item gets selected or unselected (I think the focus change is the important thing) and the TAB key gets hit. I also tested it on Windows and Linux.

**Comment by djessich** [comment]
I have the same problem using controlsfx with JavaFX 11. I tried the sample application and the problem appears there too.

**Comment by abhinayagarwal** [comment]
Interesting. This means I might be missing something. I will give it a try soon.

**Comment by abhinayagarwal** [comment]
I am still unable to reproduce the issue. Here is what I did:

1. Run ControlsFX Sampler
2. Check a random item from CheckComboBox (with data objects)
3. Press TAB moves the focus to the next CheckBox. Pressing SPACE selects the checkbox

Java Version
-----------------
openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment (build 11.0.7+10-post-Ubuntu-2ubuntu219.10)
OpenJDK 64-Bit Server VM (build 11.0.7+10-post-Ubuntu-2ubuntu219.10, mixed mode, sharing)

**Comment by djessich** [comment]
To reproduce the issue, checkout branch 9.0.0. Run with Gradle. Reproduce Steps:
Click on "Checked" -> Click on "CheckComboBox" -> Click on any CheckComboBox in opened content sample dialog -> Select an item in the open CheckComboBox using mouse click (do not close CheckComboBox) -> Press TAB on Keyboard while CheckComboBox is still opened -> Exception

**Comment by abhinayagarwal** [comment]
I still cannot reproduce the issue.

![ControlsFX-1297](https://user-images.githubusercontent.com/3197675/94360073-787f4d00-00c8-11eb-8387-381b02251104.gif)

**Comment by abhinayagarwal** [comment]
Hi @djessich ,

Since I cannot reproduce the issue, can you check if the fix in #1298 fixes the issue for you?

**Comment by djessich** [comment]
@abhinayagarwal I could successfully test this PR for branch `9.0.0`. The behavior is the same as on master branch (tested on Windows), which means that pressing TAB on the keyboard moves the focus from currently open ComboBox to the next one which will then also be opened. It does not move the focus from ComboBox item to ComboBox item (as in your showcase), which works as expected by pressing one of the arrow keys on the keyboard. The exception described by the author, is not thrown anymore.
